controllers/grid/users/chapter/ChapterGridHandler.inc.php

```diff
@@ -554,11 +553,14 @@ public function deleteChapter($args, $request)
         $chapterId = $chapter->getId();
 
         // remove Authors assigned to this chapter first
-        $chapterAuthorDao = DAORegistry::getDAO('ChapterAuthorDAO'); /* @var $chapterAuthorDao ChapterAuthorDAO */
-        $assignedAuthorIds = $chapterAuthorDao->getAuthorIdsByChapterId($chapterId);
+        $assignedAuthors = Repo::author()->getMany(
+            Repo::author()
+                ->getCollector()
+                ->filterByChapter($chapterId)
+        );
 
-        foreach ($assignedAuthorIds as $authorId) {
-            $chapterAuthorDao->deleteChapterAuthorById($authorId, $chapterId);
+        foreach ($assignedAuthors as $assignedAuthor) {
+            Repo::author()->removeFromChapter($assignedAuthor->getId(), $chapterId);
```

Comment:
Can you use the `Repo::author()->deleteChapterAuthorsByChapterId($chapterId)` here instead of getting the authors first?

Reply:
fixed